aiida/manage/configuration/profile.py

```diff
@@ -134,14 +134,27 @@ def get_repository_container(self) -> 'Container':
         """
         from disk_objectstore import Container
 
-        filepath = os.path.join(self.repository_path, 'container')
+        filepath = self._container_path
         container = Container(filepath)
 
-        if not container.is_initialised:
+        if not self.container_is_initialised:
             container.init_container(clear=True, **self.defaults['repository'])  # pylint: disable=unsubscriptable-object
 
         return container
 
+    @property
+    def container_is_initialised(self):
+        """Check if the container of the profile file repository has already been initialised."""
+        from disk_objectstore import Container
+        filepath = self._container_path
+        container = Container(filepath)
+        return container.is_initialised
```

Comment:
```suggestion
        return Container(self._container_path).is_initialised
```

Reply:
Also, if this is a more or less "constant" property, you could store it as a private variable and only create the `Container` instance once per call to this property, returning the "cached" private variable all the other times instead. I.e., you store the first result here in the private variable.